In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy.stats import pointbiserialr
from scipy.stats import pearsonr


In [2]:
datas = pd.read_csv('general_data.csv')

In [3]:
datas.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [4]:
datas.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [5]:
datas.isnull().sum()

Age                         0
Attrition                   0
BusinessTravel              0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeID                  0
Gender                      0
JobLevel                    0
JobRole                     0
MaritalStatus               0
MonthlyIncome               0
NumCompaniesWorked         19
Over18                      0
PercentSalaryHike           0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           9
TrainingTimesLastYear       0
YearsAtCompany              0
YearsSinceLastPromotion     0
YearsWithCurrManager        0
dtype: int64

In [6]:
datasnew = datas.dropna(axis=0)

In [7]:
datasnew.head(4)

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  

[4 rows x 24 columns]

In [8]:
datasnew.isnull().sum()

Age                        0
Attrition                  0
BusinessTravel             0
Department                 0
DistanceFromHome           0
Education                  0
EducationField             0
EmployeeCount              0
EmployeeID                 0
Gender                     0
JobLevel                   0
JobRole                    0
MaritalStatus              0
MonthlyIncome              0
NumCompaniesWorked         0
Over18                     0
PercentSalaryHike          0
StandardHours              0
StockOptionLevel           0
TotalWorkingYears          0
TrainingTimesLastYear      0
YearsAtCompany             0
YearsSinceLastPromotion    0
YearsWithCurrManager       0
dtype: int64

In [9]:
datasx = datasnew.copy()

In [10]:
datasx['Attrition'] = datasx['Attrition'].map({'Yes':1,'No':0})

In [11]:
datasx.head()

Age  Attrition     BusinessTravel              Department  \
0   51          0      Travel_Rarely                   Sales   
1   31          1  Travel_Frequently  Research & Development   
2   32          0  Travel_Frequently  Research & Development   
3   38          0         Non-Travel  Research & Development   
4   32          0      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                 6          2  Life Sciences              1           1   
1                10          1  Life Sciences              1           2   
2                17          4          Other              1           3   
3                 2          5  Life Sciences              1           4   
4                10          1        Medical              1           5   

   Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0  Female  ...                 1.0      Y                11              8   
1  Female  ...                 0.0      Y                23              8   
2    Male  ...                 1.0      Y                15              8   
3    Male  ...                 3.0      Y                11              8   
4    Male  ...                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [12]:
stats, p = pearsonr(datasx.Attrition,datasx.Age)
print(stats, p)

-0.1583986795409671 5.1265982193780794e-26


# Hypothesis 1- Age vs Attrition
Ho: No correlation
H1: correlation

r = -0.1583986795409671, p = 5.1265982193780794e-26

here r is -ve, it means negative correlation
p<0.05, we reject Ho & hence there is negative correlation between Age vs Attrition


In [13]:
stats, p = pearsonr(datasx.Attrition,datasx.DistanceFromHome)
print(stats, p)

-0.009448638515156248 0.5317715668019558


# Hypothesis 2- DistanceFromHome vs Attrition
Ho: No correlation
H1: correlation

r = -0.009448638515156248, p = 0.5317715668019558

here r is -ve, it means negative correlation but it is very nearer to 0.
p>0.05, we accept Ho & hence there is no correlation between Age vs Attrition


In [14]:
stats, p = pearsonr(datasx.Attrition,datasx.MonthlyIncome   )
print(stats, p)

-0.030160293808460664 0.045890862744719166


# Hypothesis 3- MonthlyIncome vs Attrition
Ho: No correlation
H1: correlation

r = -0.030160293808460664, p = 0.045890862744719166

here r is -ve, it means negative correlation.
p<0.05, we rejept Ho & hence there is negative correlation between MonthlyIncome vs Attrition


In [15]:
stats, p = pearsonr(datasx.Attrition,datasx.PercentSalaryHike)
print(stats, p)

0.03315303713546665 0.028192446935106235


# Hypothesis 4- PercentSalaryHike vs Attrition
Ho: No correlation
H1: correlation

r = 0.03315303713546665, p = 0.028192446935106235

here r is +ve, it means positive correlation.
p<0.05, we reject Ho & hence there is positive correlation between PercentSalaryHike vs Attrition


In [16]:
stats, p = pearsonr(datasx.Attrition,datasx.TrainingTimesLastYear)
print(stats, p)

-0.04758573693081737 0.0016276603635477602


# Hypothesis 5- TrainingTimesLastYear vs Attrition
Ho: No correlation
H1: correlation

r = -0.04758573693081737, p = 0.0016276603635477602

here r is -ve, it means negative correlation.
p<0.05, we reject Ho & hence there is negative correlation between TrainingTimesLastYear vs Attrition


In [17]:
stats, p = pearsonr(datasx.Attrition,datasx.YearsAtCompany)
print(stats, p)

-0.1330026184252154 9.476118084836507e-19


# Hypothesis 6- YearsAtCompany vs Attrition
Ho: No correlation
H1: correlation

r = -0.1330026184252154, p = 9.476118084836507e-19

here r is -ve, it means negative correlation.
p<0.05, we reject Ho & hence there is negative correlation between YearsAtCompany vs Attrition


In [18]:
stats, p = pearsonr(datasx.Attrition,datasx.YearsSinceLastPromotion)
print(stats, p)

-0.03142315056330995 0.03752293607395154


# Hypothesis 7- YearsSinceLastPromotion vs Attrition
Ho: No correlation
H1: correlation

r = -0.03142315056330995, p = 0.03752293607395154

here r is -ve, it means negative correlation.
p<0.05, we reject Ho & hence there is negative correlation between YearsSinceLastPromotion vs Attrition


In [19]:
stats, p = pearsonr(datasx.Attrition,datasx.YearsWithCurrManager)
print(stats, p)

-0.15469153690287274 7.105369646771178e-25


# Hypothesis 8- YearsWithCurrManager vs Attrition
Ho: No correlation
H1: correlation

r = -0.15469153690287274, p = 7.105369646771178e-25

here r is -ve, it means negative correlation.
p<0.05, we reject Ho & hence there is negative correlation between YearsWithCurrManager vs Attrition
